[Bug] Night mode - Reviewer crash

###### Reproduction Steps

1. Change to Night Mode
2. Open Reviewer 


###### Expected Result
It should let me review the cards.


###### Actual Result
The app crashed
```
2022-01-09 17:07:29.162 6133-6133/com.ichi2.anki E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.ichi2.anki, PID: 6133
    java.lang.ExceptionInInitializerError
        at com.ichi2.themes.HtmlColors.invertColors(Unknown Source:0)
        at com.ichi2.anki.cardviewer.CardHtml.getContent(CardHtml.kt:101)
        at com.ichi2.anki.cardviewer.CardHtml.getTemplateHtml(CardHtml.kt:83)
        at com.ichi2.anki.AbstractFlashcardViewer.updateCard(AbstractFlashcardViewer.java:1794)
        at com.ichi2.anki.AbstractFlashcardViewer.displayCardQuestion(AbstractFlashcardViewer.java:1677)
        at com.ichi2.anki.AbstractFlashcardViewer.displayCardQuestion(AbstractFlashcardViewer.java:1652)
        at com.ichi2.anki.Reviewer.displayCardQuestion(Reviewer.java:1163)
        at com.ichi2.anki.AbstractFlashcardViewer$NextCardHandler.onPostExecute(AbstractFlashcardViewer.java:647)
        at com.ichi2.anki.AbstractFlashcardViewer$NextCardHandler.onPostExecute(AbstractFlashcardViewer.java:592)
        at com.ichi2.anki.servicelayer.TaskListenerBuilder.replaceWith$lambda-1(AsyncLayerAdapter.kt:68)
        at com.ichi2.anki.servicelayer.TaskListenerBuilder.$r8$lambda$qteKjIdlDOFIgo-YfnGs9fMw6SA(Unknown Source:0)
        at com.ichi2.anki.servicelayer.TaskListenerBuilder$$ExternalSyntheticLambda5.accept(Unknown Source:2)
        at com.ichi2.anki.servicelayer.AsyncLayerAdapterKt$toListener$1.onPostExecute(AsyncLayerAdapter.kt:48)
        at com.ichi2.async.CollectionTask.onPostExecute(CollectionTask.java:228)
        at android.os.AsyncTask.finish(AsyncTask.java:771)
        at android.os.AsyncTask.access$900(AsyncTask.java:199)
        at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:788)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:246)
        at android.app.ActivityThread.main(ActivityThread.java:8633)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
     Caused by: java.util.regex.PatternSyntaxException: Syntax error in regexp pattern near index 47
    (.cloze\s*\{[^}]*color:\s*#)[0-9a-f]{6}(;[^}]*})
                                                   ^
        at com.android.icu.util.regex.PatternNative.compileImpl(Native Method)
        at com.android.icu.util.regex.PatternNative.<init>(PatternNative.java:39)
        at com.android.icu.util.regex.PatternNative.create(PatternNative.java:35)
        at java.util.regex.Pattern.compile(Pattern.java:1426)
        at java.util.regex.Pattern.<init>(Pattern.java:1401)
        at java.util.regex.Pattern.compile(Pattern.java:985)
        at com.ichi2.themes.HtmlColors.<clinit>(HtmlColors.kt:37)
        	... 23 more
2022-01-09 17:07:29.163 6133-6133/com.ichi2.anki D/UsageAnalytics: sendAnalyticsException() description/fatal: java.util.regex.PatternSyntaxException: Syntax error in regexp pattern near index 47
    (.cloze\s*\{[^}]*color:\s*#)[0-9a-f]{6}(;[^}]*})
                                                   ^/true
2022-01-09 17:07:29.167 6133-6133/com.ichi2.anki I/ACRA: ACRA is disabled for com.ichi2.anki - forwarding uncaught Exception on to default ExceptionHandler
```

###### Debug info
AnkiDroid Version = 2.16alpha35

Android Version = 11

Manufacturer = samsung

Model = SM-M307F

Hardware = exynos9611

Webview User Agent = Mozilla/5.0 (Linux; Android 11; SM-M307F Build/RP1A.200720.012; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/97.0.4692.70 Mobile Safari/537.36

ACRA UUID = 7c517bc5-27de-4634-bbfe-e5e939d20d8a

Scheduler = std2

Crash Reports Enabled = false

DatabaseV2 Enabled = true

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

--- a/AnkiDroid/src/main/java/com/ichi2/themes/HtmlColors.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/themes/HtmlColors.kt
@@ -19,7 +19,6 @@ package com.ichi2.themes
 import com.ichi2.utils.HashUtil.HashMapInit
 import com.ichi2.utils.KotlinCleanup
 import timber.log.Timber
-import java.lang.NumberFormatException
 import java.util.*
 import java.util.regex.Matcher
 import java.util.regex.Pattern
@@ -33,8 +32,8 @@ object HtmlColors {
     private val fLongHexColorPattern = Pattern.compile("^#([0-9a-f]{2})([0-9a-f]{2})([0-9a-f]{2})$", Pattern.CASE_INSENSITIVE)
     private val fRgbColorPattern = Pattern.compile("^rgb\\(([0-9]+)\\s*,\\s*([0-9]+)\\s*,\\s*([0-9]+)\\)$", Pattern.CASE_INSENSITIVE)
 
-    // In Android, } should be escaped
-    private val fClozeStylePattern = Pattern.compile("(.cloze\\s*\\{[^}]*color:\\s*#)[0-9a-f]{6}(;[^}]*})", Pattern.CASE_INSENSITIVE)
+    @Suppress("RegExpRedundantEscape") // In Android, } should be escaped
+    private val fClozeStylePattern = Pattern.compile("(.cloze\\s*\\{[^}]*color:\\s*#)[0-9a-f]{6}(;[^}]*\\})", Pattern.CASE_INSENSITIVE)
     private fun nameToHex(name: String): String? {
         if (sColorsMap == null) {
             sColorsMap = HashMapInit(fColorsRawList.size)

